utilities/object_registry.cc

```diff
@@ -185,5 +207,29 @@ void ObjectRegistry::Dump(Logger *logger) const {
   }
 }
 
+Status ObjectRegistry::RegisterPlugin(const std::string &name,
+                                      const PluginFunc &plugin_func) {
+  Plugin plugin;
+  std::string errmsg;
+  int code = plugin_func(&plugin, &errmsg);
+  if (code != 0) {  // TODO: Perhaps use different codes?
```

Comment:
If we were ever to support dynamic plugins, Status could break ABI.  

Additionally, some of the other functions in the ObjectRegistry (like NewFactoryObject) follow the same model (of returning something and an error string.

It would be possible to use the Status::Code enums as return codes (hence the TODO) to have this method return different Status for different cases (instead of always returning InvalidArgument).

Reply:
> If we were ever to support dynamic plugins, Status could break ABI.

What use is a RocksDB plugin that is ABI-incompatible with the compiled RocksDB?

Reply:
> 

A Plugin that is not ABI-compatible is not good, which is why there used to be code in the Plugin that tried to detect that condition and prevented it from loading.  If a Plugin was not ABI-compatible and returned a Status here that was not ABI-compatible, it could cause the program to crash and would not be easy to debug or decipher the problem.